[DOCS] Adding docs for foreign-data wrappers (#11888)
yugabyte/yugabyte-db: docs/content/latest/explore/ysql-language-features/pg-extensions.md

## Using PostgreSQL extensions

### file_fdw example

> Firstly, install the extension:

Reviewer: ```suggestion
First, install the extension:
```

```sql
CREATE EXTENSION file_fdw;
```

> Create a foreign server:

Reviewer: Is the term "foreign server" correct here? Or should we say "Create a server from a foreign data wrapper"?

Author: Maybe `create a foreign server using the foreign data wrapper` ? or is that redundant 
either way, "foreign server" is in line with PG terminology

Reviewer: If it's standard terminology, it's all good. Let's leave well enough alone. :)

```sql
CREATE SERVER my_server FOREIGN DATA WRAPPER file_fdw;
```

> Now, you can create foreign tables that access data from files.

Reviewer: ```suggestion
Now, you can create foreign tables that access data from files. For example:
```

```sql
CREATE FOREIGN TABLE employees (id int, employee_name varchar) SERVER myserver OPTIONS (filename 'employees.csv', format 'csv');
```

> `SELECT`s can be executed on the created foreign tables to access the data in the corresponding files.

Reviewer: ```suggestion
You can execute `SELECT` statements on the foreign tables to access the data in the corresponding files.
```

The `crosstab()`and  `crosstabN()` functions produce "pivot" displays. The _"N"_ in crosstabN() indicates the fact that a few, `crosstab1()`, `crosstab2()`, `crosstab3()`, are provided natively by the extension and that you can follow documented steps to create more.

### postgres_fdw example

> Firstly, install the extension:

Reviewer: ```suggestion
First, install the extension:
```

```sql
CREATE EXTENSION postgres_fdw;
```

> To connect to a remote postgres / YSQL database, create a foreign server object. The connection information (except the username and password) can be specified using the `OPTIONS` clause.

Reviewer: ```suggestion
To connect to a remote YSQL or PostgreSQL database, create a foreign server object. Specify the connection information (except the username and password) using the `OPTIONS` clause:
```

```sql
CREATE SERVER my_server FOREIGN DATA WRAPPER postgres_fdw OPTIONS (host 'host_ip', dbname 'external_db', port 'port_number');
```

> The username and password should be specified using `CREATE USER MAPPING`.

Reviewer: ```suggestion
Specify the username and password using `CREATE USER MAPPING`:
```

```sql
CREATE USER MAPPING FOR mylocaluser SERVER myserver OPTIONS (user 'remote_user', password 'password');
```

You can now create foreign tables using `CREATE FOREIGN TABLE` and `IMPORT FOREIGN SCHEMA`.

```sql
CREATE FOREIGN TABLE table_name (colname1 int, colname2 int) SERVER myserver OPTIONS (schema_name 'schema', table_name 'table');

> ```
> 
> ```sql

Reviewer: I'd just combine these two
```suggestion
```

IMPORT FOREIGN SCHEMA foreign_schema_name FROM SERVER my_server INTO local_schema_name;
```

> `SELECT`s can be executed on the created foreign tables to access the data in the corresponding remote tables.

Reviewer: ```suggestion
You can execute `SELECT` statements on the foreign tables to access the data in the corresponding remote tables.
```
